### PR TITLE
Upgrade SDK to use v5 API.

### PR DESCRIPTION
**Pull request by relango**

### CODE REVIEW

File: src/main/java/com/salesforce/einsteinbot/sdk/client/model/BotHttpHeaders.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+  public Map<String, Collection<String>> getAll() {
+    return headerValues.asMap();
+  }
+
+  public Collection<String> get(String headerName) {
+    return headerValues.get(headerName);
+  }
+
+  public MultiValueMap<String, String> toMultiValueMap() {
```
Comment (atulkc) [inline src/main/java/com/salesforce/einsteinbot/sdk/client/model/BotHttpHeaders.java:85]: Why do we need to use spring framework's `MultiValueMap` when we are also using `com.google.common.collect.Multimap`?

File: src/main/java/com/salesforce/einsteinbot/sdk/client/model/BotHttpHeaders.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        .ofNullable(headerValues.get(headerName))
+        .flatMap(this::findFirstItem);
+  }
+
+  private Optional<String> findFirstItem(Collection<String> collection) {
+    return collection.stream().findFirst();
+  }
+
+  public Map<String, Collection<String>> getAll() {
+    return headerValues.asMap();
```
Comment (atulkc) [inline src/main/java/com/salesforce/einsteinbot/sdk/client/model/BotHttpHeaders.java:78]: Should return `Collections.unmodifiableMap()`

**Review by atulkc — COMMENTED** [review]
Overall looks good. Few minor comments inline.

File: src/main/java/com/salesforce/einsteinbot/sdk/client/model/BotHttpHeaders.java (hunk 2 above)
Comment (relango) [reply src/main/java/com/salesforce/einsteinbot/sdk/client/model/BotHttpHeaders.java:78]: Sure. will fix

File: src/main/java/com/salesforce/einsteinbot/sdk/cache/Cache.java [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 /**
  * Cache interface provides an abstraction for plugging in a cache mechanism for maintaining the
- * session cache. This interface is used by SessionManagedChatbotClient to manage the mapping
+ * session cache. This interface is used by SessionManagedChatbotClientImpl to manage the mapping
```
[the remark this answers is not in the document]
Comment (relango) [reply src/main/java/com/salesforce/einsteinbot/sdk/cache/Cache.java:14]: will update the comments

File: pom.xml [hunk 4]
```diff
@@ -39,8 +39,8 @@
         <spring-reactor-test-version>3.2.3.RELEASE</spring-reactor-test-version>
         <spring-reactor-core-version>3.4.5</spring-reactor-core-version>
         <openapi-generator-version>5.4.0</openapi-generator-version>
-        <runtime-api-spec-url>https://runtime-api-na-east.prod.chatbots.sfdc.sh/docs/specs/${api-spec-yaml-file}</runtime-api-spec-url>
-        <api-spec-yaml-file>v4_0_0_api_specs.yml</api-spec-yaml-file>
+        <runtime-api-spec-url>https://runtime-api-na-west.stg.chatbots.sfdc.sh/docs/specs/${api-spec-yaml-file}</runtime-api-spec-url>
```
[the remark this answers is not in the document]
Comment (relango) [reply pom.xml:42]: will add a todo until we update it.

**Review by atulkc — APPROVED** [review]
LGTM 👍

**Pull request merged**